src/GoogleCloudStorageServiceProvider.php

```diff
@@ -78,6 +76,28 @@ public function boot()
         });
     }
 
+    /**
+     * Create a new StorageClient
+     *
+     * @param  mixed $config
+     * @return \Google\Cloud\Storage\StorageClient
+     */
+    private function createClient($config)
+    {
+        if (is_string($config['key_file'])) {
+            return new StorageClient([
+                'projectId' => $config['project_id'],
+                'keyFilePath' => $config['key_file'],
+            ]);
+        }
+
+        return new StorageClient([
+            'projectId' => $config['project_id'],
+            'keyFile' => array_merge([
+                "project_id" => $config['project_id']
+            ], $config['key_file'])
```

Comment:
This will cause an exception if `key_file` is not set, or is not an array..
```suggestion
            ], array_get($config, 'key_file', [])
```